src/Employer/Employer.Web/Controllers/VacancyPreviewController.cs

```diff
@@ -50,7 +50,11 @@ public async Task<IActionResult> Submit(SubmitEditModel m)
 
             if (ModelState.IsValid)
             {
-                return RedirectToRoute(RouteNames.Submitted_Index_Get);
+                if (response.Data.IsSubmitted)
+                    return RedirectToRoute(RouteNames.Submitted_Index_Get);
+
+                if (response.Data.HasLegalEntityAgreement == false)
+                    return RedirectToRoute(RouteNames.LegalEntityAgreement_HardStop_Get);
```

Comment:
There is a flag redundancy as I see it, IsSubmitted and HasLegalAgreement they both reflect to the same cause... this can be changed to ... 

```suggestion
                if (response.Data.HasLegalEntityAgreement == false)
                    return RedirectToRoute(RouteNames.LegalEntityAgreement_HardStop_Get);
                
                return RedirectToRoute(RouteNames.Submitted_Index_Get);    
```

Reply:
My thinking is this is more explicit.